### DISCUSSION TITLE
Full backup budibase without cli

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#8128] [Help]
Hi all,

We have a self hosted version of Budibase installed with docker-compose and .env file without cli. 
Is there a way to have a full system backup similarly the way is done with the cli?
Is it possible to install and use cli with an budibase existing installation?

**Comment by melohagan** [comment]
Hey @gmentzik 

You should be able to use the Budibase CLI with an existing installation. 

First step would be to backup your docker-compose and .env files if you have custom config you want to keep.

Next run `budi hosting --init` and follow through on the docs: https://docs.budibase.com/docs/budibase-cli-setup

This will create a new docker-compose file and .env file which you can input any custom config as needed. 

Then you can follow through the backup instructions: https://docs.budibase.com/docs/backups

**Reply by pbarmak** [reply]
@melohagan, I have a follow-up question on the backups. You mention using the CLI backup (which is new to me, thanks for that link!). Prior to seeing that, I was told to use the backup script from here: https://github.com/thomaskamalakis/budibasebackup.

Using the cli approach you mention, I get a relatively small backup file (in my case, around 800K). Using the other python script, I get backups with much larger sizes (in the 15M range).

I understand the python version seems to be copying full containers, so is it maybe backing up way more than we need? Before I switch over to the CLI version for our backups, I just wanted to make sure that the CLI version backs up everything we need. I know I should test a restore myself to be certain, but thought I'd ask the experts while I set that up.

**Reply by melohagan** [reply]
Yeah the CLI backup will be a lot more efficient, as it only needs to backup the Budibase data (users, screens, internal DB rows), and not the image data. It also zips everything in a tarball.

**Reply by gmentzik** [reply]
@melohagan I have few questions
So whether you have a running system it is recommended to use: budi hosting --init , to download everything from scratch and then modify the docker compose and env file according to your demands. Since I am using the official docker-compose file just adding a mysql container and some minor changes is --init necessary? What other files does it download to your folder.
Secondly I understand from the documentation that in order to restore you must have a clean Budibase installation otherwise conficts may occur. So does this mean that for daily backups is better to also keep a copy of budibase docker volumes?

**Reply by melohagan** [reply]
You only need `budi hosting --init` when you first setup Budibase. From that point on use `budi hosting --update` to pull in the latest images.

Frequent backups will always provide a shorter fallback window. It really depends on how frequently you're changing and updating your apps - every day seems a little excessive. Maybe you can backup before making big changes or making a big update.
